pkg/util/crypt_linux.go

```diff
@@ -0,0 +1,43 @@
+// +build linux
+
+package util
+
+import (
+	"unsafe"
+)
+
+/*
+#cgo LDFLAGS: -lcrypt
+#define _GNU_SOURCE
+#include <crypt.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+char *do_gnu_source_crypt_r(const char *key, const char *salt) {
+	struct crypt_data data = { 0, };
+	char *result = crypt_r(key, salt, &data);
+	if (!result)
+		return NULL;
```

Comment:
From the documentation of cgo, it would appear that ```nil, nil``` would never be returned. Nevertheless, since this is binding "unsafe" code better to be safe. Added a check.